rhino/src/main/java/org/mozilla/javascript/JavaMembers.java

```diff
@@ -595,23 +599,21 @@ private void reflect(
                     NativeJavaMethod setters = null;
                     String setterName = "set".concat(nameComponent);
 
-                    if (ht.containsKey(setterName)) {
```

Comment:
I think we can safely remove `containsKey` here, as we call `get` in the next line, which is checked for null by the `instanceOf` test.